### PR TITLE
Simplify behavior on invalid content-type

### LINKED ISSUE #130: Redefinition of HTTP: Content-Type on PUT
```
If not exactly one Content-Type header was received as part of a
PUT request or the Content-Type header value contains non-ASCII
characters, or it is unreasonably long, the server MAY refuse to
process the request, and instead respond with a descriptive error
message in the body, and a http response code from the 4xx range.
```

In particular:

> or the Content-Type header value contains non-ASCII characters

We're redefining HTTP here again.

https://tools.ietf.org/html/rfc7231#section-3.1.1.5 defines an ABNF for the header value, anything outside of that char range is invalid and _obviously_ triggers a 4xx since it's _syntactically_ invalid
#125 tracks moving this whole paragraph to "best-practices" if I understood correctly, but atm I think it's best to fully remove this paragraph.

### PR DESCRIPTION
**Pull request by untitaker**
Fix #130

### CODE REVIEW

File: source.txt [hunk 1]
```diff
@@ -246,11 +246,10 @@ Table of Contents
...
 
-    If not exactly one Content-Type header was received as part of a
-    PUT request, the Content-Type header value contains non-ASCII
-    characters, or it is unreasonably long, the server MAY refuse to
-    process the request, and instead respond with a descriptive error
-    message in the body, and a http response code from the 4xx range.
+    If no valid Content-Type header was received as part of a PUT
+    request, the server MAY refuse to process the request, and instead
+    respond with a descriptive error message in the body, and a http
+    response code from the 4xx range.
```
Comment (skddc) [inline source.txt]: 1. Just a proposal to improve the form: "and instead respond with a descriptive error message in the body, as well as an HTTP response code from the 4xx range."
2. It's not entirely clear who or what defines "valid" here.

Comment (untitaker) [inline source.txt]: 1. agreed
2. The entire header, and therefore its validity, is defined in the HTTP RFC

**Comment by skddc** [comment]
If two of us think it's not clear enough, then why not clarify it now? To me it's not even clear who or what defines "valid" and my comment wasn't addressed before merging.

**Comment by untitaker** [comment]
Imo it is reasonable to assume (as you do) that if the RS spec doesn't explicitly place restrictions on the content-type, that only the ones from the HTTP RFC apply. Though it probably makes sense to literally say that no further restrictions apply.

**Comment by untitaker** [comment]
Also @skddc this PR is about removing a restriction that is already found in the HTTP spec. I think we do agree that this is a good thing.

**Comment by skddc** [comment]
Ok, so if I understand this correctly, the best way to clarify this would be to add the respective section of the HTTP RFC as an annotation there.

**Comment by untitaker** [comment]
#152

**Comment by michielbdejong** [comment]
We should update the changelog btw, since this is breaking for clients. They now have to pick their content-types more carefully than in the -07 spec. It makes the protocol a lot less easy to define. First of all, when a new media-type is added to the official registry, should servers start accepting that on the same day? or within a month? It's just not clear whether clients can safely use content-types that were only added recently. Also, it's not clear from the http spec if content-types starting with `x-` are officially allowed, or whether that's just an undocumented hack that exists only in practice (or maybe I didn't read far enough).

The spec is slowly moving from an API definition (stating exactly what happens in which case) to an internet spec, where a range of behaviors is allowed in various degrees. I guess this is a good thing, but it's not always practical. ;)

**Comment by untitaker** [comment]
x-types are definetly allowed. I don't see the ambiguity or the backwards compat breakage for clients that followed the HTTP spec to begin with.

**Comment by michielbdejong** [comment]
> for clients that followed the HTTP spec

yeah, fair point :)

**Comment by michielbdejong** [comment]
I just found out that this was merged into master, but not included in the -08 build. I'll include it in the -09 build.

**Pull request merged**